packages/devtools_app/lib/src/shared/editor/api_classes.dart

```diff
@@ -31,14 +31,36 @@ enum EditorMethod {
 /// TODO(https://github.com/flutter/devtools/issues/8824): Add tests that these
 /// are in-sync with analysis_server.
 enum LspMethod {
+  codeAction(methodName: 'textDocument/codeAction'),
   editableArguments(methodName: 'dart/textDocument/editableArguments'),
-  editArgument(methodName: 'dart/textDocument/editArgument');
+  editArgument(methodName: 'dart/textDocument/editArgument'),
+  executeCommand(methodName: 'workspace/executeCommand');
 
   const LspMethod({required this.methodName});
 
+  /// Returns the [LspMethod] for the given [methodName].
+  ///
+  /// If the [methodName] does not exist, returns null.
+  static LspMethod? fromMethodName(String methodName) {
+    for (final method in LspMethod.values) {
+      if (method.methodName == methodName) return method;
+    }
+    return null;
```

Comment:
can this be `return LspMethod.values.firstWhereOrNull((method) => method.methodName == methodName)`

Reply:
Done! Though we need the **method** name map instead of the name map, so I've added a static map for method name -> method to use for the look-up.